File: 1_Two_Sum/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :return type: List[int]
        """
        hash = dict()
        for i in range(len(nums)):
            complement = target - nums[i]
            if (complement in hash):
                return [hash[complement], i]
            hash[nums[i]] = i
        
    

        

File: 1_Two_Sum/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let dict = {}
    for (i = 0; i<nums.length; i++){
        
        if ((target - nums[i]) in dict){
            return [dict[target - nums[i]], i]
        } else{
            dict[nums[i]] = i
        }
    }
};
